sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala

```diff
@@ -68,17 +68,17 @@ object ResolveInlineTables extends Rule[LogicalPlan]
   /**
    * Validates that all inline table data are valid expressions that can be evaluated
    * (in this they must be foldable).
-   *
+   * Exception are CURRENT_LIKE expressions, which are replaced by a literal in later stages.
    * This is package visible for unit testing.
    */
   private[analysis] def validateInputEvaluable(table: UnresolvedInlineTable): Unit = {
     table.rows.foreach { row =>
       row.foreach { e =>
-        // Note that nondeterministic expressions are not supported since they are not foldable.
-        // Only exception are CURRENT_LIKE expressions, which are replaced by a literal
-        // In later stages.
-        if ((!e.resolved && !e.containsPattern(CURRENT_LIKE))
-          || !trimAliases(prepareForEval(e)).foldable) {
+        if (e.containsPattern(CURRENT_LIKE)) {
+          return
```

Comment:
nit: we should not use `return` in a lambda, as non-local return is harmful in general. To simulate this early return code style, we can do
```
if (e.containsPattern(CURRENT_LIKE)) {
  // do nothing
} else if (...) ...
```